memory/libhugetlbfs.py

```diff
@@ -89,24 +89,36 @@ def setUp(self):
         else:
             self.cancel("Kernel does not support hugepages")
 
-        if not self.hugetlbfs_dir:
-            self.hugetlbfs_dir = os.path.join(self.teststmpdir, 'hugetlbfs')
-            os.makedirs(self.hugetlbfs_dir)
+        self.configured_page_sizes = []
+        self.hugetlbfs_dir = {}
 
         for hp_size in self.page_sizes:
-            if process.system('mount -t hugetlbfs -o pagesize=%sM none %s' %
-                              (hp_size, self.hugetlbfs_dir), sudo=True,
-                              ignore_status=True):
-                self.cancel("hugetlbfs mount failed")
-            genio.write_file(
-                '/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages' %
-                str(int(hp_size) * 1024), str(pages_requested))
+            try:
+                genio.write_file(
+                    '/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages' %
+                    str(int(hp_size) * 1024), str(pages_requested))
+            except OSError:
+                if (int(hp_size) * 1024) == 16777216:
+                    self.log.warn('Running 16GB hugepages')
```

Comment:
why not cancel here as well

Reply:
This change is to not cancel and continue with the tests and **16G** is not configurable with Hash page table.